subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -318,6 +318,11 @@ static int on_publish_evt(struct mqtt_client *const client,
 	bool doc_update_rejected =
 		aws_jobs_cmp(update_topic, topic, topic_len, "rejected");
 
+#if defined(CONFIG_AWS_FOTA_LOG_LEVEL_DBG)
+	/* Fix for log print out when debugging */
+	u8_t *topic_in = (u8_t *)topic;
+	topic_in[topic_len] = '\0';
```

Comment:
the input buffer should not be modified beyond its specified length, even if you do cast away the const.

Reply:
That's exactly what this does. It modifies the last character in the specified length, isn't it? But I do agree with both comments that it shouldn't be modified. My first iteration was making a local copy then using that as the input for the log function I guess I'll revert to that.

Reply:
`topic_len` is not a string length, it is the length of the buffer.
so `topic_in[topic_len-1]` would access the last bye of `topic`

Reply:
but that would be the last valid character meaning you can't really overwrite it to get a terminated string, but I'll change the code :)